File: android/app/src/main/java/com/smartexpense/android/presentation/util/ThemeManager.java
package com.smartexpense.android.presentation.util;

import android.content.Context;
import android.content.SharedPreferences;
import androidx.core.content.ContextCompat;
import com.smartexpense.android.R;

public class ThemeManager {

    private static final String PREF_NAME = "set_theme_prefs";
    private static final String KEY_ACCENT_COLOR = "selected_accent_color";

    public enum AccentColor {
        NEON_LIME(R.color.accent_neon_lime, "Xanh Neon"),
        NEON_CYAN(R.color.accent_neon_cyan, "Xanh Cyan"),
        NEON_YELLOW(R.color.accent_neon_yellow, "Vàng Neon"),
        NEON_PINK(R.color.accent_neon_pink, "Hồng Neon"),
        NEON_PURPLE(R.color.accent_neon_purple, "Tím Neon"),
        NEON_ORANGE(R.color.accent_neon_orange, "Cam Neon"),
        NEON_MINT(R.color.accent_neon_mint, "Xanh Mint"),
        NEON_CORAL(R.color.accent_neon_coral, "Đỏ Neon");

        private final int colorResId;
        private final String displayName;

        AccentColor(int colorResId, String displayName) {
            this.colorResId = colorResId;
            this.displayName = displayName;
        }

        public int getColorResId() {
            return colorResId;
        }

        public String getDisplayName() {
            return displayName;
        }
    }

    public static void setAccentColor(Context context, AccentColor color) {
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        prefs.edit().putString(KEY_ACCENT_COLOR, color.name()).apply();
    }

    public static AccentColor getAccentColor(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        String name = prefs.getString(KEY_ACCENT_COLOR, AccentColor.NEON_YELLOW.name());
        try {
            return AccentColor.valueOf(name);
        } catch (IllegalArgumentException e) {
            return AccentColor.NEON_YELLOW;
        }
    }

    public static int getAccentColorInt(Context context) {
        AccentColor current = getAccentColor(context);
        return ContextCompat.getColor(context, current.getColorResId());
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/presentation/util/ThemeManager.kt
package com.smartexpense.android.presentation.util

import android.content.Context
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import com.smartexpense.android.ui.theme.*

object ThemeManager {
    private const val PREF_NAME = "set_theme_prefs"
    private const val KEY_ACCENT_COLOR = "selected_accent_color"

    enum class AccentColor(val composeColor: Color, val displayName: String, val startColor: Color, val endColor: Color) {
        SUNRISE(ThemeSunriseStart, "Bình Minh", ThemeSunriseStart, ThemeSunriseEnd),
        SUNSET(ThemeSunsetStart, "Hoàng Hôn", ThemeSunsetStart, ThemeSunsetEnd),
        BERRY(ThemeBerryStart, "Mâm Xôi", ThemeBerryStart, ThemeBerryEnd),
        OCEAN(ThemeOceanStart, "Đại Dương", ThemeOceanStart, ThemeOceanEnd),
        FOREST(ThemeForestStart, "Rừng Xanh", ThemeForestStart, ThemeForestEnd),
        LIME(ThemeLimeStart, "Chanh Tươi", ThemeLimeStart, ThemeLimeEnd),
        FLAMINGO(ThemeFlamingoStart, "Hồng Hạc", ThemeFlamingoStart, ThemeFlamingoEnd),
        AMETHYST(ThemeAmethystStart, "Thạch Anh", ThemeAmethystStart, ThemeAmethystEnd),
        FIRE(ThemeFireStart, "Lửa Đỏ", ThemeFireStart, ThemeFireEnd),
        TROPICAL(ThemeTropicalStart, "Nhiệt Đới", ThemeTropicalStart, ThemeTropicalEnd),
        AURORA(ThemeAuroraStart, "Cực Quang", ThemeAuroraStart, ThemeAuroraEnd),
        NEON_NIGHT(ThemeNeonNightStart, "Đêm Neon", ThemeNeonNightStart, ThemeNeonNightEnd);

        val composeBrush: Brush
            get() = Brush.linearGradient(listOf(startColor, endColor))
    }

    fun setAccentColor(context: Context, color: AccentColor) {
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_ACCENT_COLOR, color.name).apply()
    }

    fun getAccentColor(context: Context): AccentColor {
        val prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
        val name = prefs.getString(KEY_ACCENT_COLOR, AccentColor.SUNRISE.name)
            ?: AccentColor.SUNRISE.name
        return try {
            AccentColor.valueOf(name)
        } catch (e: IllegalArgumentException) {
            AccentColor.SUNRISE
        }
    }

    fun getAccentComposeColor(context: Context): Color {
        return getAccentColor(context).composeColor
    }
}
